test/int.jl

```diff
@@ -192,6 +197,32 @@ for T in (Int8, Int16, Int32, Int64, Int128)
     @test_throws OverflowError checked_mul(sqrt2, -sqrt2)
     @test_throws OverflowError checked_mul(-sqrt2, sqrt2)
     @test_throws OverflowError checked_mul(-sqrt2, -sqrt2)
+
+    diverr = WORD_SIZE == 32 && T === Int128 ? InexactError : DivideError
```

Comment:
On 32-bit systems, some Int128 arithmetic is implemented via BigInt to circumvent LLVM bugs. `int.jl` describes this well in comments. The respective conversion from `-big(typemin(Int128))` to `Int128` throws an `InexactError`. I think it should throw an `OverflowError` instead of an `InexactError`. I also think that  `div(typemin(Int128), Int128(-1))` should throw an `OverflowError` instead of a `DivideError` (see #14273).

All in all, I decided to just document current behaviour instead of trying to fix it here. Changing the Int128 work-around code to throw a `DivideError` would be straightforward.